File: 03-strings/py/strings.py
if __name__ == "__main__":
    homer = "I want a donut"
    marge = homer.replace("donut", "kiss")

    print(f"Homer says : {homer}")
    print(f"Marge replies : {marge} instead")
    print(f"Marge adds : {marge} please")

    phrase_homer = "trying is the first step towards failure"
    for word in reversed(phrase_homer.split()):
        print(">", word)

    chars = [c for c in phrase_homer]
    chars = sorted(set(chars))

    s = ""
    for c in chars:
        s += c
        s += ", "

    trim = ' ,'
    trimmed_str = s.strip(trim)
    print("Used    chars:", s)
    print("Trimmed chars:", trimmed_str)

File: 03-strings/rs/src/strings.rs
pub fn demo() {

    println!("===");
    println!("Demo strings");
    println!("===");

    let homer = String::from("I want a donut");
    let marge: String = homer.replace("donut", "kiss");

    println!("Homer says: {}", homer);
    println!("Marge replies: {}", marge);
    println!("Marge adds: {} please", marge);

    let phrase_homer: &'static str = "trying is the first step towards failure";
    println!("Homer's quote: {}", phrase_homer);

    println!("Words in reverse");
    for word in phrase_homer.split_whitespace().rev() {
        println!("> {}", word);
    }

    let mut chars: Vec<char> = phrase_homer.chars().collect();
    chars.sort();
    chars.dedup();

    let mut s = String::new();
    for c in chars {
        s.push(c);
        s.push_str(", ");
    }

    let trim: &[char] = &[' ', ','];
    let trimmed_str: &str = s.trim_matches(trim);
    println!("Used    characters: {}", s);
    println!("Trimmed characters: {}", trimmed_str);
}